src/main/java/dev/blaauwendraad/masker/json/SegmentReference.java

```diff
@@ -0,0 +1,20 @@
+package dev.blaauwendraad.masker.json;
+
+/**
+ * A mutable reference to a sequence of bytes in <code>message</code>. It is used to represent json path segments.
+ * A "key" segment type reference is represented as a (start, offset) pair.
+ * For an "array index" segment type reference, a (start, offset) pair is interpreted as (index, -1).
+ */
+public class SegmentReference {
```

Comment:
Can't this be called `JsonPathSegmentReference` or just `JsonPathSegment`? 

1 year from now I will not remember that SegmentReference is about JsonPath segments

Reply:
`JsonPathSegmentReference` is a more suitable name indeed. I will rename it.